### PR TITLE
fix(model details): remove redundant table header

### LINKED ISSUE #486: Relation provider table isn't full width (2.8)
image

### PR DESCRIPTION
**Pull request by barrymcgee**
@hatched I removed this header as there is no content being generated for the column - correct?

## Done

(fix) Remove redundant table header
(fix) Correct first letter capitalisation util 

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- View model details page for model with relations info
- View relations table and verify headers match columns

## Details

Fixes #486

### CODE REVIEW

**Comment by hatched** [comment]
There can be....are you saying you've found a spec that removes this data from Juju?

**Comment by barrymcgee** [comment]
@hatched I'm saying no logic exists to generate this column data so I'm asking if that reason is that it's unavailable to us?

**Comment by hatched** [comment]
Ahh, no this data is potentially available, so if it's not provided lets just default to an empty string instead of removing the column.

File: src/pages/Models/Details/generators.js [hunk 1]
```diff
@@ -42,7 +42,6 @@ export const relationTableHeaders = [
   { content: "requirer" },
   { content: "interface" },
   { content: "type" },
-  { content: "message" },
```
Comment (hatched) [inline src/pages/Models/Details/generators.js:45]: Lets just default this to an empty string if we do not have data to render, as the content may be available.

Comment (barrymcgee) [reply src/pages/Models/Details/generators.js:45]: Given we have limited screen real estate, particularly on smaller screens, I don't think we should be rendering columns of whitespace. When we have the data, then we render the column?

**Comment by spencerbygraves** [comment]
@hatched I think @barrymcgee is right, can we use the space and show the column when we have the data.

@clagom would you agree?

**Comment by clagom** [comment]
I think this can work nicely, something we can iterate on, but generally speaking would help to display the data, yes @spencerbygraves

**Comment by hatched** [comment]
This is an architectural change to how the tables operate. We haven't currently hidden any columns if we do not have content for them. If that's something we want to implement then it's a much bigger task than this. We would probably want some way to fade in/out columns as the data arrives/leaves. This would also reflow the table, potentially while the user is using it, I'm not sure I've ever seen a tabular data representation with dynamic columns.

**Comment by barrymcgee** [comment]
@hatched We appear to be talking at cross purposes - let's discuss in standup.

**Comment by clagom** [comment]
I thought the question was 'let's do it' or 'not doing it' meaning something trivial. But if requires more than this then definitely it's not high priority

**Comment by barrymcgee** [comment]
https://github.com/canonical-web-and-design/jaas-dashboard/issues/504

**Pull request merged**